java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -625,6 +635,21 @@ private predicate argToQualifierStep(Expr tracked, Expr sink) {
     tracked = ma.getArgument(i) and
     sink = ma.getQualifier()
   )
+  or
+  exists(Method m, MethodAccess ma |
+    taintPreservingArgumentToQualifier(m) and
+    ma.getMethod() = m and
+    tracked = ma.getAnArgument() and
+    sink = ma.getQualifier()
+  )
+}
+
+/**
+ * Holds if `method` is a method that transfers taint from any of its arguments to its qualifier.
+ */
+private predicate taintPreservingArgumentToQualifier(Method method) {
+  method instanceof StringFormatMethod and
+  not method.getDeclaringType() instanceof TypeString
```

Comment:
Perhaps state that method is not `static` instead of having a special case for `TypeString`